File: laravel/app/Models/SyncOperation.php
<?php

declare(strict_types=1);

namespace App\Models;

use App\Models\Concerns\HasUuidPrimaryKey;
use Illuminate\Database\Eloquent\Model;

class SyncOperation extends Model
{
    use HasUuidPrimaryKey;

    protected $table = 'sync_operations';

    public $incrementing = false;

    protected $keyType = 'string';

    protected $fillable = [
        'client_id',
        'device_id',
        'operation_id',
        'entity_type',
        'entity_id',
        'operation_type',
        'payload',
        'client_created_at',
        'received_at',
        'processed_at',
        'status',
        'retry_count',
        'error_code',
        'error_message',
    ];

    protected $casts = [
        'id' => 'string',
        'entity_id' => 'string',
        'payload' => 'array',
        'client_created_at' => 'datetime',
        'received_at' => 'datetime',
        'processed_at' => 'datetime',
        'retry_count' => 'integer',
    ];
}

File: docs/flutter/lib/core/sync/sync_operation.dart
final class SyncOperation {
  const SyncOperation({
    required this.operationId,
    required this.deviceId,
    required this.entityType,
    required this.entityId,
    required this.operationType,
    required this.clientCreatedAt,
    required this.payload,
    this.clientSequence,
  });

  final String operationId;
  final String deviceId;
  final String entityType;
  final String entityId;
  final String operationType;
  final DateTime clientCreatedAt;
  final int? clientSequence;
  final Map<String, dynamic> payload;
}
